### PR TITLE
Update pex_reactor.go

### PR DESCRIPTION
**Pull request by lcm16001080**
Resolve IP bug in public network。

Please add a description of the changes that this PR introduces and the files that
are the most critical to review.

If this PR fixes an open Issue, please include "Closes #XXX" (where "XXX" is the Issue number) 
so that GitHub will automatically close the Issue when this PR is merged.

### CODE REVIEW

File: p2p/pex/pex_reactor.go [hunk 1]
```diff
@@ -205,7 +205,7 @@ func (r *Reactor) AddPeer(p Peer) {
 			r.Logger.Error("Failed to get peer NetAddress", "err", err, "peer", p)
 			return
 		}
-
+		addr.IP = p.SocketAddr().IP
```
Comment (alexanderbez) [inline p2p/pex/pex_reactor.go:208]: Hey @lcm16001080, mind adding a bit more context here? What is this change and why is it necessary?

Comment (lcm16001080) [reply p2p/pex/pex_reactor.go:208]: When we use it on the public network, we find that addr.Ip is the internal network address.
When the other node initiates the brush bucket, the addr.Ip directly responds to the request, resulting in a connection error.
So here I change addr.Ip to the external network address.

Comment (alexanderbez) [reply p2p/pex/pex_reactor.go:208]: Does this change make sense @melekes @erikgrinaker?

Comment (erikgrinaker) [reply p2p/pex/pex_reactor.go:208]: I think this is too simplistic. For example, the remote operator may have set an appropriate advertised address via `external_address`, which we'd be replacing here based on a guess. Also, even though we know the IP address, we're just guessing that the remote port would be 26657, and that the port will be open and forwarded to the node.

This is a known problem, as address detection and advertisement is pretty broken at the moment, and should be fixed properly in the new P2P stack. See e.g. #5588.

Comment (melekes) [reply p2p/pex/pex_reactor.go:208]: agree w/ Erik. @lcm16001080 do u want to help us with a proper fix?

Comment (lcm16001080) [reply p2p/pex/pex_reactor.go:208]: Addr.IP = p.SocketAddr().IP 
The line of code I submitted should fix the #5588 problem.

Comment (lcm16001080) [reply p2p/pex/pex_reactor.go:208]: Compare the data before and after AddrBook. json repair, see the IP data.
1. Fix the data before AddrBook. json

{
	"key": "0c34a9aaf341a03a1b2d13ef",
	"addrs": [
		{
			"addr": {
				"id": "d592a6593c18fa1d511986c4a9763c150a2c2f02",
				"ip": "0.0.0.0",
				"port": 26306
			},
			"src": {
				"id": "d592a6593c18fa1d511986c4a9763c150a2c2f02",
				"ip": "0.0.0.0",
				"port": 26306
			},
			"buckets": [
				87
			],
			"attempts": 0,
			"bucket_type": 1,
			"last_attempt": "2020-11-12T15:27:01.554575+08:00",
			"last_success": "0001-01-01T00:00:00Z"
		},
		{
			"addr": {
				"id": "f0feb21e845198b036de1c4d6df02c93cbc351e9",
				"ip": "0.0.0.0",
				"port": 26406
			},
			"src": {
				"id": "f0feb21e845198b036de1c4d6df02c93cbc351e9",
				"ip": "0.0.0.0",
				"port": 26406
			},
			"buckets": [
				87
			],
			"attempts": 3,
			"bucket_type": 1,
			"last_attempt": "2020-11-12T16:27:46.535264+08:00",
			"last_success": "0001-01-01T00:00:00Z"
		},
		{
			"addr": {
				"id": "d4ec4fe638ff365326e8dafb39a8e6c555eb85f2",
				"ip": "0.0.0.0",
				"port": 26206
			},
			"src": {
				"id": "d4ec4fe638ff365326e8dafb39a8e6c555eb85f2",
				"ip": "0.0.0.0",
				"port": 26206
			},
			"buckets": [
				87
			],
			"attempts": 0,
			"bucket_type": 1,
			"last_attempt": "2020-11-12T15:26:55.113754+08:00",
			"last_success": "0001-01-01T00:00:00Z"
		}
	]
}

2. After fixing the AddrBook. json data
config$ cat addrbook.json
{
	"key": "c1c896ee2c3d0c2214e45641",
	"addrs": [
		{
			"addr": {
				"id": "3d1de247b1c96e62e607ca4d713fa2da8e7381ec",
				"ip": "18.162.72.189",
				"port": 26656
			},
			"src": {
				"id": "3d1de247b1c96e62e607ca4d713fa2da8e7381ec",
				"ip": "18.162.72.189",
				"port": 26656
			},
			"buckets": [
				49
			],
			"attempts": 0,
			"bucket_type": 2,
			"last_attempt": "2021-03-11T00:50:28.227357689Z",
			"last_success": "2021-03-11T00:50:28.227357689Z"
		},
		{
			"addr": {
				"id": "e942087a716d00f403f068b4358a9a507eacec59",
				"ip": "18.162.204.60",
				"port": 26656
			},
			"src": {
				"id": "e942087a716d00f403f068b4358a9a507eacec59",
				"ip": "18.162.204.60",
				"port": 26656
			},
			"buckets": [
				31
			],
			"attempts": 0,
			"bucket_type": 2,
			"last_attempt": "2021-03-11T00:50:17.38157533Z",
			"last_success": "2021-03-11T00:50:17.38157533Z"
		},
		{
			"addr": {
				"id": "f909b49af1cfdfe52c959611306dd0f556fc39e1",
				"ip": "18.166.219.32",
				"port": 26656
			},
			"src": {
				"id": "f909b49af1cfdfe52c959611306dd0f556fc39e1",
				"ip": "18.166.219.32",
				"port": 26656
			},
			"buckets": [
				49
			],
			"attempts": 0,
			"bucket_type": 2,
			"last_attempt": "2021-03-11T00:19:06.694530943Z",
			"last_success": "2021-03-11T00:19:06.694530943Z"
		},
		{
			"addr": {
				"id": "2ea51638e8beb34bf76e90961a6cd9b556a1719c",
				"ip": "18.166.131.97",
				"port": 26656
			},
			"src": {
				"id": "2ea51638e8beb34bf76e90961a6cd9b556a1719c",
				"ip": "18.166.131.97",
				"port": 26656
			},
			"buckets": [
				35
			],
			"attempts": 0,
			"bucket_type": 2,
			"last_attempt": "2021-03-11T00:55:20.974386581Z",
			"last_success": "2021-03-11T00:55:20.974386581Z"
		}
	]
}

Comment (lcm16001080) [reply p2p/pex/pex_reactor.go:208]: @brapse @husio @xla @svenstaro @davebryson Please check, thank you.

**Pull request closed without merging**